pkg/watcher/reconciler/dynamic/dynamic.go

```diff
@@ -396,6 +499,13 @@ func (r *Reconciler) streamLogs(ctx context.Context, o results.Object, logType,
 		Err: inMemWriteBufferStderr,
 	}, logChan, errChan)
 
+	// pull the first error that occurred and return on that; reminder - per https://golang.org/ref/spec#Channel_types
+	// channels act as FIFO queues
+	chanErr, ok := <-errChan
```

Comment:
since @khrm last review I've added here a simplified version of the error handling that @sayan-biswas used to do prior to pr #712 

let's just return the first error seen; from what I see in the tkn code, multiple errors are not collected

with this though @sayan-biswas and the prior changes made with #712, plus moving the `sendLogs` call back on the reconciler thread, we now address one of the short comings of the old implementation, namely ignoring of errors and preventing retries on reconciliation.

Reminder:  with the `threadiness` arg, users can adjust the number of threads when they enable log support.

I will add a separate PR to add the k8s client tuning shortly

Reply:
https://github.com/tektoncd/results/pull/744 is the add tuning options PR